### ISSUE TITLE
Incorrect URL for microservice swagger

### ISSUE THREAD

**Issue by chrisipeters** [Burgyn/MMLib.SwaggerForOcelot#64]
**Describe the bug**
Swagger launched from the gateway is looking for the microservice swagger json at the wrong path

**Expected behavior**
Swagger should populate microservice json data

**To Reproduce**
Gateway API: localhost:44393
Users Microservice: localhost:44340

I am able to access the Users microservice json file directly at: https://localhost:44340/swagger/v1/swagger.json

But from the gateway page I am getting the error:
**Fetch error: Internal Server Error /swagger/docs/v1/users**

I'm unable to figure out why that path is being used rather than the one specified in SwaggerEndPoints:Config:Url

**In the Gateway API**
ocelot.json file:

```
{
  "ReRoutes": [
    {
      "UpstreamPathTemplate": "/users/{everything}",
      "DownstreamPathTemplate": "/{everything}",
      "SwaggerKey": "users",
      "DownstreamScheme": "https",
      "DownstreamHostAndPorts": [
        {
          "Host": "localhost",
          "Port": 44340
        }
      ]
    }
  ],
  "SwaggerEndPoints": [
    {
      "Key": "users",
      "Config": [
        {
          "Name": "Users API",
          "Version": "v1",
          "Url": "https://localhost:44340/swagger/v1/swagger.json"
        }
      ]
    }
  ],
  "GlobalConfiguration": {
    "BaseUrl": "https://localhost:44393"
  },
  "AuthenticationOptions": {
    "authPersistSingleRequest": "True"
  }
}

```

gateway startup file:

```
using System.Collections.Generic;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using Ocelot.DependencyInjection;
using Ocelot.Middleware;

namespace endpointApi.gateway
{
    public class Startup
    {
        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }
        public IConfiguration Configuration { get; }

        // This method gets called by the runtime. Use this method to add services to the container.
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddOcelot();
            services.AddSwaggerForOcelot(Configuration);
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }

            app.UseStaticFiles();

            app.UseAuthorization();

            app.UseSwaggerForOcelotUI(Configuration, opt =>
            {
                opt.DownstreamSwaggerHeaders = new[]
                {
                        new KeyValuePair<string, string>("Key", "Value"),
                        new KeyValuePair<string, string>("Key2", "Value2"),
                    };
            })
                .UseOcelot()
                .Wait();
        }
    }
}
```

** Users microservice Startup**

```
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using Microsoft.OpenApi.Models;

namespace endpointApi.sites.users
{
    public class Startup
    {
        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }
        public IConfiguration Configuration { get; }

        public void ConfigureServices(IServiceCollection services)
        {
            services.AddControllers();

            services.AddMvc();

            services.AddSwaggerGen(c =>
            {
                c.SwaggerDoc("v1", new OpenApiInfo { Title = "Users API", Version = "v1" });
            });
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }

            app.UseSwagger()
                .UseSwaggerUI(c =>
                {
                    c.SwaggerEndpoint("/swagger/v1/swagger.json", "Users API V1");
                });

            app.UseRouting();

            app.UseAuthorization();


            app.UseEndpoints(endpoints =>
            {
                endpoints.MapControllers();
            });
        }
    }
}
```

**Comment by Burgyn** [comment]
Hi,

please can you try to replace `{catchAll}` with `{everything}` in `ocelot.json`?

**Comment by chrisipeters** [comment]
ocelot.json updated as above, error now reads:
Fetch error: undefined /swagger/docs/v1/users

**Comment by Burgyn** [comment]
Hi, 
Can there be a problem with certificate?

What if you change `SwaggerEndPoints.Url` to `http` instead of `https`. For test only.

**Comment by emrehorasan** [comment]
Hi, 

I have same problem too. 'try execute' button is making a call to `http://http://localhost:5000/bro..` adress

**Comment by Burgyn** [comment]
Hi @emrehorasan,
your issue looks different. Befacause you have duplicated `http` in path. It looks like a problem with `baseUrl`. Please create new issue for this and attach your `ocelot.json` and downstream swagger json.

thanks.

**Comment by Burgyn** [comment]
Hi @emrehorasan,

your issue looks like #53. 

As a workaround you can downgrade to version `1.8`

**Comment by emrehorasan** [comment]
Thanks for your quick answer @Burgyn 
 
I cannot downgrade to version 1.8 because I using .net core 3.1. Incompatibility occurs in other packages.

**Comment by Burgyn** [comment]
Hi @emrehorasan 
Issue #53 is fixed in version `1.10.2`
